lib/policies/load-balancing.js

```diff
@@ -319,39 +336,32 @@ TokenAwareIterator.prototype.iterate = function (callback) {
 };
 
 TokenAwareIterator.prototype.computeNext = function () {
-  //return local replicas
-  var host = this.nextLocalReplica();
-  if (host) {
-    return {value: host, done: !host};
+  var host;
+  if (this.localReplicaIndex < this.localReplicas.length) {
+    host = this.localReplicas[(this.startIndex + (this.localReplicaIndex++)) % this.localReplicas.length];
+    return { value: host, done: false };
   }
-  //return child hosts
-  var item = this.childIterator.next();
-  if (!item.done) {
+  this.startIndex = -1;
```

Comment:
Good question, I went for that approach initially but it that causes uneven distribution of the replicas in the case the amount remote and local replicas differ, ie: 

- local replicas: 3; remote: 6; possible start indexes: `[0, 1, 2]`.

If we use a `rnd * total replicas`, we still can have an undesired distribution, ie: 

- local replicas 4; remote 6; possible start indexes [0...9], that for local replicas when applying `mod` translates to `[0, 1, 2, 3, 0, 1, 2, 3, 0, 1]`.